controllers/prep_handlers.go

```diff
@@ -173,53 +169,170 @@ func (r *ImageBasedUpgradeReconciler) queryPrecachingStatus(ctx context.Context,
 	return
 }
 
-func (r *ImageBasedUpgradeReconciler) launchSetupStateroot(
-	ibu *lcav1alpha1.ImageBasedUpgrade, progressfile string) (result ctrl.Result, err error) {
-	if err = os.Chdir("/"); err != nil {
-		return
+func (r *ImageBasedUpgradeReconciler) SetupStateroot(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) error {
+	r.Log.Info("Start setupstateroot")
+
+	workspaceOutsideChroot, err := os.MkdirTemp(common.PathOutsideChroot("/var/tmp"), "")
+	if err != nil {
+		return fmt.Errorf("failed to create temp directory %w", err)
 	}
 
-	// Write the script
-	scriptname := filepath.Join(utils.IBUWorkspacePath, utils.PrepSetupStateroot)
-	scriptcontent, _ := generated.Asset(utils.PrepSetupStateroot)
-	err = os.WriteFile(common.PathOutsideChroot(scriptname), scriptcontent, 0o700)
+	defer func() {
+		if err := os.RemoveAll(workspaceOutsideChroot); err != nil {
+			r.Log.Error(err, "failed to clean up "+workspaceOutsideChroot)
+		}
+		if _, err := r.Executor.Execute("podman", "rmi", ibu.Spec.SeedImageRef.Image); err != nil {
+			r.Log.Error(err, "failed to remove image"+ibu.Spec.SeedImageRef.Image)
+		}
+	}()
 
+	workspace, err := filepath.Rel(utils.Host, workspaceOutsideChroot)
 	if err != nil {
-		err = fmt.Errorf("failed to write handler script: %s, %w", common.PathOutsideChroot(scriptname), err)
-		return
+		return fmt.Errorf("failed to get workspace relative path %w", err)
 	}
+	r.Log.Info("workspace:" + workspace)
 
-	go r.Executor.Execute(scriptname, "--seed-image", ibu.Spec.SeedImageRef.Image, "--progress-file", progressfile,
-		"--os-version", ibu.Spec.SeedImageRef.Version, "--os-name", getStaterootName(ibu.Spec.SeedImageRef.Version))
+	if _, err = r.Executor.Execute("mount", "/sysroot", "-o", "remount,rw"); err != nil {
+		return fmt.Errorf("failed to remount /sysroot: %w", err)
+	}
 
-	result = requeueWithShortInterval()
+	mountpoint, err := r.Executor.Execute("podman", "image", "mount", ibu.Spec.SeedImageRef.Image)
+	if err != nil {
+		return fmt.Errorf("failed to mount seed image: %w", err)
+	}
 
-	return
-}
+	defer func() {
+		if _, err := r.Executor.Execute("podman", "image", "umount", ibu.Spec.SeedImageRef.Image); err != nil {
```

Comment:
same question as with rmi, if it will fail here, we will not be able to run it one more time, right?

Reply:
no this only logs an error. It won't fail the prep stage.

Reply:
should it fail the stage?

Reply:
We should probably fail in this scenario.